pkgs/development/python-modules/argilla/default.nix

```diff
@@ -0,0 +1,106 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, deprecated
+, rich
+, backoff
+, packaging
+, pydantic
+, typer
+, tqdm
+, wrapt
+, numpy
+, httpx
+, pandas
+, monotonic
+# test dependencies
+, pytestCheckHook
+, fastapi
+, sqlalchemy
+, opensearch-py
+, factory_boy
+, elasticsearch8
+, elastic-transport
+, luqum
+, pytest-asyncio
+, passlib
+, python-jose
+, alembic
+, uvicorn
+}:
+let
+  pname = "argilla";
+  version = "1.8.0";
+in
+buildPythonPackage {
+  inherit pname version;
+  format = "setuptools";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "argilla-io";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-pUfuwA/+fe1VVWyGxEkvSuJLNxw3sHmp8cQZecW8GWY=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace '"rich <= 13.0.1"' '"rich"' \
+      --replace '"numpy < 1.24.0"' '"numpy"'
+  '';
+
+  propagatedBuildInputs = [
+    deprecated
+    rich
+    backoff
+    packaging
+    pydantic
+    typer
+    tqdm
+    wrapt
+    numpy
+    pandas
+    httpx
+    monotonic
+  ];
+
+  preCheck = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  doCheck = false;
+  # ERROR tests/client/test_api.py
+  # ERROR tests/client/test_dataset.py
+  # ERROR tests/client/test_feedback.py
+  # ERROR tests/functional_tests/test_log_for_token_classification.py
+  # ERROR tests/labeling/text_classification/test_label_errors.py
+  # ERROR tests/listeners/test_listener.py - RuntimeError: Failed to import argilla.listeners.listener becaus>
+  # ERROR tests/server/datasets/test_dao.py - argilla.server.daos.backend.base.GenericSearchError: Connection>
+  # ERROR tests/server/info/test_api.py
```

Comment:
Those lines are not really helpful. Either you post the complete test errors or not.